File: apps/frontend/src/api/models/galleryStages.ts
export interface GalleryStageConfig {
  stage_index: number;
  title: string;
  description: string;
  scene_description: string;
  tags: string[];
  video_prompt: string;
}

export interface GalleryStagesConfigResponse {
  influencer_id: string;
  character_id: number;
  character_slug: string;
  source: string;
  stages: GalleryStageConfig[];
  default_stages: GalleryStageConfig[];
}

export interface GalleryStagesConfigPayload {
  stages: GalleryStageConfig[];
}

export function buildStageVideoPrompt(stage: GalleryStageConfig): string {
  const videoPrompt = stage.video_prompt.trim();
  if (videoPrompt) {
    return videoPrompt;
  }
  return (
    `Create a natural looping video variation for stage ${stage.stage_index}. ` +
    "Subtle idle motion that repeats seamlessly."
  );
}

File: apps/backend/app/data/schemas/gallery_stages.py
from pydantic import BaseModel, Field


class GalleryStageConfigOut(BaseModel):
    stage_index: int
    title: str
    description: str = ""
    scene_description: str = ""
    tags: list[str] = []
    video_prompt: str = ""


class GalleryStagesConfigOut(BaseModel):
    influencer_id: str
    character_id: int
    character_slug: str
    source: str = "character_default"
    stages: list[GalleryStageConfigOut]
    default_stages: list[GalleryStageConfigOut] = []


class GalleryStageConfigIn(BaseModel):
    stage_index: int = Field(ge=1, le=5)
    title: str = ""
    description: str = ""
    scene_description: str = ""
    tags: list[str] = []
    video_prompt: str = ""


class GalleryStagesConfigIn(BaseModel):
    stages: list[GalleryStageConfigIn]
